### DISCUSSION TITLE
What happened to CatalogIndex module?

### DISCUSSION THREAD

**Discussion by kiatng** [OpenMage/magento-lts#3405] [Questions & Answers] [answered]
There is no table associated to CatalogIndex in the db. And it seems the following always returns `false`:

https://github.com/OpenMage/magento-lts/blob/875661e3c23cdc465ae088ab7a73da23346afdb3/app/code/core/Mage/CatalogIndex/Model/Aggregation.php#L45-L48

So, `$data` is always `null` here:

https://github.com/OpenMage/magento-lts/blob/875661e3c23cdc465ae088ab7a73da23346afdb3/app/code/core/Mage/Catalog/Model/Layer/Filter/Attribute.php#L111-L113

Is the CatalogIndex module dead?

**Comment by addison74** [comment]
I can't give you an answer. Is it a feature that has worked in the past? If so, then we have to check where we lost it on the way.

**Comment by fballiano** [comment]
what table are you searching for, so I'll check some old projects I have

**Comment by kiatng** [comment]
The tables are defined in the config.xml and created in the sql directory but in the latest version it's completely missing.

**Comment by fballiano** [comment]
@kiatng it is something that should stop us from releasing?

**Comment by kiatng** [comment, marked as answer]
No. I now think we should remove the module , may be in the `next` branch.

**Comment by sreichel** [comment]
@kiatng not tested ... is it possbly used with (non-)flat tables?

**Reply by kiatng** [reply]
I have both _Use Flat Catalog Category_ and _Use Flat Catalog Product_ set to _No_. No evidence that that `CatalogIndex` is used.

**Reply by fballiano** [reply]
I've enable both flat catalog and add some "die" to some catalogIndex files, it seems nothing from that module is run

**Reply by fballiano** [reply]
I've also removed all PHP files from that module, nothing seem to break in reindex phase, with or without flat.

**Reply by sreichel** [reply]
Mhh ... did you test it with active cart/catalog rules? My last idea where it could be used ...

**Reply by fballiano** [reply]
tested with catalog/cart rules, can't find any broken reindex. I'll do a PR on the next branch.

**Comment by fballiano** [comment]
Note for future self, this is also related to https://github.com/OpenMage/magento-lts/pull/2758
